src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs

```diff
@@ -795,109 +795,258 @@ public static unsafe int IndexOfAny(ref char searchSpace, char value0, char valu
 
-                    while (length > 0)
+        VectorizedScan:
+            // We get past SequentialScan only if IsHardwareAccelerated or intrinsic .IsSupported is true. However, we still have the redundant check to allow
+            // the JIT to see that the code is unreachable and eliminate it when the platform does not have hardware accelerated.
+            if (Avx2.IsSupported)
+            {
+                lengthToExamine = GetCharVector256SpanLength(offset, length);
+                if (lengthToExamine > 0)
+                {
+                    Debug.Assert(length - offset >= Vector256<ushort>.Count);
+                    Vector256<ushort> values0 = Vector256.Create(value0);
+                    Vector256<ushort> values1 = Vector256.Create(value1);
+                    Vector256<ushort> values2 = Vector256.Create(value2);
+                    Vector256<ushort> values3 = Vector256.Create(value3);
+                    Vector256<ushort> values4 = Vector256.Create(value4);
+                    do
                     {
-                        // Using Unsafe.Read instead of ReadUnaligned since the search space is pinned and pCh is always vector aligned
-                        Debug.Assert(((int)pCh & (Unsafe.SizeOf<Vector<ushort>>() - 1)) == 0);
-                        Vector<ushort> vData = Unsafe.Read<Vector<ushort>>(pCh);
-                        var vMatches = Vector.BitwiseOr(
-                                            Vector.BitwiseOr(
-                                                Vector.BitwiseOr(
-                                                    Vector.BitwiseOr(Vector.Equals(vData, values0), Vector.Equals(vData, values1)),
-                                                    Vector.Equals(vData, values2)),
-                                                Vector.Equals(vData, values3)),
-                                            Vector.Equals(vData, values4));
-
-                        if (Vector<ushort>.Zero.Equals(vMatches))
+                        Vector256<ushort> search = LoadVector256(ref searchStart, offset);
+                        // We preform the Or at non-Vector level as we are using the maximum number of non-preserved registers (+ 1),
+                        // and more causes them first to be pushed to stack and then popped on exit to preseve their values.
+                        int matches = Avx2.MoveMask(Avx2.CompareEqual(values0, search).AsByte());
+                        // Bitwise Or to combine the flagged matches for the second, third, fourth and fifth values to our match flags
+                        matches |= Avx2.MoveMask(Avx2.CompareEqual(values1, search).AsByte());
+                        matches |= Avx2.MoveMask(Avx2.CompareEqual(values2, search).AsByte());
+                        matches |= Avx2.MoveMask(Avx2.CompareEqual(values3, search).AsByte());
+                        matches |= Avx2.MoveMask(Avx2.CompareEqual(values4, search).AsByte());
```

Comment:
Should be faster to `Or` all the `CompareEqual` results together and do a single `MoveMask` at the end.  `vpmovmskb` has 2-3 cycles latency and can only be scheduled on a single port.  Latency is even greater on Zen.

Reply:
Never mind.  Didn't read the first sentence in the comment :/